lm_eval/api/model.py

```diff
@@ -296,11 +297,18 @@ def _encode_pair(self, context, continuation):
             continuation = context[-n_spaces:] + continuation
             context = context[:-n_spaces]
 
-        whole_enc = self.tok_encode(context + continuation)
-        context_enc = self.tok_encode(context)
+        if self.AUTO_MODEL_CLASS == transformers.AutoModelForCausalLM:
+            whole_enc = self.tok_encode(
+                context + continuation, add_special_tokens=False
```

Comment:
no need for add_special_tokens to be hardcoded in _encode_pair, right? tok_encode sets add_special_tokens based on arch / based on `add_bos_token` flag